static/src/javascripts/projects/common/modules/commercial/slice-adverts.js

```diff
@@ -35,27 +35,38 @@ define([
                 return false;
             }
 
-            var container, containerId, $adSlice, isFrontFirst,
+            var container, containerId, $adSlice, isFrontFirst, fabricAdSlot,
                 opts = defaults(
                     options || {},
                     {
                         containerSelector: '.fc-container',
                         sliceSelector: '.js-fc-slice-mpu-candidate'
                     }
                 ),
-                // get all the containers
+            // get all the containers
                 containers   = qwery(opts.containerSelector),
                 index        = 0,
                 adSlices     = [],
                 containerGap = 1,
-                prefs        = userPrefs.get('container-states');
+                prefs        = userPrefs.get('container-states'),
+                addFabricAd  = (config.page.isFront && config.switches.fabricAdverts && detect.isBreakpoint({max : 'phablet'}));
+
+            // insert fabric advert at first container in lieu of a top slot
+            if (addFabricAd) {
+                fabricAdSlot = bonzo(createAdSlot('fabric', 'container-inline'));
+                fabricAdSlot.addClass('ad-slot--mobile');
+                fabricAdSlot.insertAfter(containers[0]);
+            }
+
+            // skip the initial containers if we've already added an advert to the first
+            index+= addFabricAd ? (1 + containerGap) : 0;
```

Comment:
Trivial: you could move that statement inside the `if` and remove the ternary operator.